### ISSUE TITLE
Seeded randomness in tests is not isolated, so a result depends on collection order

### ISSUE THREAD

**Issue by TheWisp** [TheWisp/lerobot#163]
`tests/hvla/test_ball_cue.py::test_cue_dropout_replaces_the_right_share_and_only_while_training`
passes on its own and passes when `tests/hvla` is run as a directory, and fails
when the full suite runs:

```
# pytest tests/hvla/test_ball_cue.py                      -> 25 passed
# pytest tests/hvla                                       -> passes
# pytest tests --ignore=tests/policies                    -> FAILS
```

It fails identically at `ball-pre-rebase-2026-08-28`, so this is not caused by
the rebase of the branch stack; it is the tree's own behaviour.

The test seeds and then asserts a *proportion*:

```python
torch.manual_seed(0)
model.train()
model.encode_observations(dict(batch))
rate = model._dropped / model._drop_seen
assert 0.42 < rate < 0.58, f"realised dropout {rate:.3f} is not the requested 0.5"
```

`torch.manual_seed(0)` reseeds the global generator, so the draw sequence should
be fixed from that line — unless something an earlier test left behind changes
which generator is consumed or how much RNG is drawn before the dropout mask.
Global default device and default dtype are both candidates, and both are
process-wide and not restored by any fixture.

The cost is not this one test. A suite where a seeded assertion depends on
collection order cannot be trusted to attribute a failure: when this failed
after a large rebase, deciding whether the rebase had broken cue dropout took a
full-suite run against the pre-rebase tree to settle, because no smaller
reproduction exists. Anything statistical in the suite has the same exposure.

Related: #84 is the same shape for `test_overlays_lifecycle_e2e` (passes
isolated, fails under full-suite load) but by resource contention rather than
RNG state. The body of #151 records a third instance among the stateless
contract tests.
